### DISCUSSION TITLE
How to parse C source from python script

### DISCUSSION THREAD

**Discussion by Hanndel** [NationalSecurityAgency/ghidra#5140] [Q&A]
Basically the title, I found the CParser constructor but cannot find something to actually insert it into the file.

cpar = CParser(open("E:\Descargas\Il2CppDumper\il2cpp_ghidra.h", 'r'))

**Comment by giuliano-macedo** [comment]
You can use [CParserUtils.parseHeaderFiles](https://ghidra.re/ghidra_docs/api/ghidra/app/util/cparser/C/CParserUtils.html#parseHeaderFiles(ghidra.program.model.data.DataTypeManager%5B%5D,java.lang.String%5B%5D,java.lang.String%5B%5D,java.lang.String,ghidra.util.task.TaskMonitor))

```python
from ghidra.app.util.cparser.C import CParserUtils

file_path = "E:\Descargas\Il2CppDumper\il2cpp_ghidra.h"
CParserUtils.parseHeaderFiles([], [file_path], [], [], currentProgram.getDataTypeManager(), monitor)
```

**Reply by ReversingWithMe** [reply]
```java
        DataTypeManager dtManager = currentProgram.getDataTypeManager();
        CParser parser = new CParser(dtManager);

        snippet_of_c from somewhere 
        
        DataType dataType = parser.parse(snippet_of_c);
```

is how I have done something similar, I expect this is exposed in python as well.
